### DISCUSSION TITLE
New to Homebrew and Terminal, need help....

### DISCUSSION THREAD

**Discussion by BadRobot-cooked** [Homebrew/discussions#1023] [Getting started]
I have a MacMini 2018 and wanted to install SmartMonTools. (I already installed in on MBA M1 to see SSD data)
I thought I had installed Brew already, but not SmartMonTools. So went ahead and entered "brew install smartmontools"
Some activity went by, then an error message...

**The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink share/doc/smartmontools/AUTHORS
Target /usr/local/share/doc/smartmontools/AUTHORS
already exists. You may want to remove it:
rm '/usr/local/share/doc/smartmontools/AUTHORS'**

So I thought this meant smartmoxntools was already present. But entering command "smartctl -a disk0 " gets the reply smartctl not found.

I then thought, OK uninstall brew....brew uninstall
Which gives this message, after a bit of text activity

**Error: Invalid usage: This command requires at least 1 installed formula or installed cask argument.**

I have no idea what to do next, not a terminal expert!

**Comment by carlocab** [comment]
Try `brew link --overwrite smartmontools`. As the flag suggests, it'll overwrite anything it conflicts with in `/usr/local`. If you want to see what'll be overwritten before actually doing it, you can do `brew link --dry-run smartmontools`.

**Comment by BadRobot-cooked** [comment]
this is what it replies...

Error: Could not symlink share/doc/smartmontools/AUTHORS
/usr/local/share/doc/smartmontools is not writable.

**Reply by carlocab** [reply]
What's the output of
```
ls -la /usr/local
ls -la /usr/local/share
```
?

**Reply by BadRobot-cooked** [reply]
wow, quick reply and thank you! it's a long reply...
total 0
drwxr-xr-x   16 root     wheel    512  8 Mar 23:39 .
drwxr-xr-x@  11 root     wheel    352  1 Jan  2020 ..
-rw-r--r--@   1 root     wheel      0 27 Sep  2019 .com.apple.installer.keep
drwxrwxr-x    2 macmini  admin     64 12 Sep  2019 Caskroom
drwxrwxr-x   70 macmini  admin   2240  8 Mar 19:02 Cellar
drwxrwxr-x    4 macmini  admin    128  7 May  2020 Frameworks
drwxrwxr-x   21 macmini  admin    672  8 Mar 19:13 Homebrew
drwxrwxr-x  243 macmini  admin   7776 11 Mar 16:50 bin
drwxrwxr-x   12 macmini  admin    384  9 Mar 15:41 etc
drwxrwxr-x  116 macmini  admin   3712  7 May  2020 include
drwxrwxr-x@ 457 root     admin  14624  7 May  2020 lib
drwxrwxr-x   79 macmini  admin   2528 11 Mar 16:50 opt
drwxr-xr-x@   3 root     wheel     96 12 Apr  2019 remotedesktop
drwxrwxr-x   13 macmini  admin    416  9 Mar 15:39 sbin
drwxrwxr-x   25 macmini  admin    800  7 May  2020 share
drwxrwxr-x    6 macmini  admin    192  8 Mar 19:02 var
total 0
drwxrwxr-x   25 macmini  admin   800  7 May  2020 .
drwxr-xr-x   16 root     wheel   512  8 Mar 23:39 ..
-rw-r--r--    1 macmini  admin     0  7 May  2020 .keepme
drwxr-xr-x   23 macmini  admin   736  7 May  2020 aclocal
lrwxr-xr-x    1 macmini  admin    43 12 Sep  2019 bash-completion -> ../Cellar/glib/2.62.0/share/bash-completion
drwxr-xr-x   28 macmini  admin   896  7 May  2020 doc
lrwxr-xr-x    1 macmini  admin    37 12 Sep  2019 ffmpeg -> ../Cellar/ffmpeg/4.1.4_2/share/ffmpeg
drwxr-xr-x    3 macmini  admin    96 12 Sep  2019 fish
lrwxr-xr-x    1 macmini  admin    44 12 Sep  2019 fontconfig -> ../Cellar/fontconfig/2.13.1/share/fontconfig
lrwxr-xr-x    1 macmini  admin    31 12 Sep  2019 gdb -> ../Cellar/glib/2.62.0/share/gdb
drwxr-xr-x   12 macmini  admin   384  7 May  2020 gettext
lrwxr-xr-x    1 macmini  admin    47  7 May  2020 gettext-0.20.2 -> ../Cellar/gettext/0.20.2_1/share/gettext-0.20.2
lrwxr-xr-x    1 macmini  admin    38 12 Sep  2019 gir-1.0 -> ../Cellar/harfbuzz/2.6.1/share/gir-1.0
lrwxr-xr-x    1 macmini  admin    36 12 Sep  2019 glib-2.0 -> ../Cellar/glib/2.62.0/share/glib-2.0
lrwxr-xr-x    1 macmini  admin    42 12 Sep  2019 graphite2 -> ../Cellar/graphite2/1.3.13/share/graphite2
drwxr-xr-x    3 macmini  admin    96 12 Sep  2019 gtk-doc
drwxr-xr-x   32 macmini  admin  1024  7 May  2020 info
drwxr-xr-x    4 macmini  admin   128 12 Sep  2019 java
drwxr-xr-x  104 macmini  admin  3328  7 May  2020 locale
drwxr-xr-x    7 macmini  admin   224 29 Jan  2020 man
lrwxr-xr-x    1 macmini  admin    34  7 May  2020 misc -> ../Cellar/libmagic/5.38/share/misc
drwxr-xr-x    3 root     admin    96  9 Mar 15:39 smartmontools
lrwxr-xr-x    1 macmini  admin    40 12 Sep  2019 tessdata -> ../Cellar/tesseract/4.1.0/share/tessdata
lrwxr-xr-x    1 macmini  admin    37 12 Sep  2019 xml -> ../Cellar/fontconfig/2.13.1/share/xml
drwxr-xr-x    3 macmini  admin    96 12 Sep  2019 zsh

**Reply by carlocab** [reply]
This looks like your problem:
```
drwxr-xr-x 3 root admin 96 9 Mar 15:39 smartmontools
```
It's owned by `root`. Compare this to all the other entries, which indicate they are owned by `macmini` (I'm assuming you get `macmini` as the output of `whoami`.)

I'm guessing you've installed `smartmontools` outside of Homebrew before. To fix this, just remove the offending `smartmontools` directory with
```
sudo rm -rf /usr/local/share/doc/smartmontools
```
The `sudo` command will ask for your password. Once you've done that, try
```
brew reinstall smartmontools
```

**Reply by BadRobot-cooked** [reply]
there is still an error message in amongst the reply, so not sure if smartmontools is actually re-installed...
==> Downloading https://homebrew.bintray.com/bottles/smartmontools-7.2.big_sur.b
Already downloaded: /Users/macmini/Library/Caches/Homebrew/downloads/2a4e1036f7ac31b5abc450600b8bab6db0ef1d43618e96013cf18cba7f4b838c--smartmontools-7.2.big_sur.bottle.tar.gz
==> Reinstalling smartmontools 
==> Pouring smartmontools-7.2.big_sur.bottle.tar.gz
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink share/man/man5/smartd.conf.5
Target /usr/local/share/man/man5/smartd.conf.5
already exists. You may want to remove it:
  rm '/usr/local/share/man/man5/smartd.conf.5'

To force the link and overwrite all conflicting files:
  brew link --overwrite smartmontools

To list all files that would be deleted:
  brew link --overwrite --dry-run smartmontools

Possible conflicting files are:
/usr/local/share/man/man5/smartd.conf.5
/usr/local/share/man/man8/smartctl.8
/usr/local/share/man/man8/smartd.8
/usr/local/share/man/man8/update-smart-drivedb.8
/usr/local/share/smartmontools/drivedb.h
==> Summary
🍺  /usr/local/Cellar/smartmontools/7.2: 36 files, 1.9MB

**Reply by carlocab** [reply]
You'll need to remove the rest of the conflicting files. Do
```
sudo rm -rf /usr/local/share/man/man5/smartd.conf.5
sudo rm -rf /usr/local/share/man/man8/smartctl.8
sudo rm -rf /usr/local/share/man/man8/smartd.8
sudo rm -rf /usr/local/share/man/man8/update-smart-drivedb.8
sudo rm -rf /usr/local/share/smartmontools/drivedb.h
brew reinstall smartmontools
```

**Reply by BadRobot-cooked** [reply]
genius! now it works and thank you I'd never have done that myself.
